File: 0169-majority-element/0169-majority-element.cpp
class Solution {
public:
    int majorityElement(vector<int>& nums) {
        if(nums.size() == 1){
            return nums[0];
        }
        int n = nums.size()-1;
        int count = 0;
        int candidate;

        for(int i=0;i<nums.size();i++){
            if(count == 0){
                candidate = nums[i];
                count++;
            }
            else if(candidate == nums[i]){
                count++;
            }
            else{
                count--;
            }
            
        }
        count =0;
        for(int i=0;i<nums.size();i++){
            if(candidate == nums[i]){
                count++;
            }
        }
        if(count >nums.size()/2){
            return candidate;
        }
        return 0;
    }
};

File: 0169-majority-element/0169-majority-element.py
class Solution(object):
    def majorityElement(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        map1 = {}

        size =0
        if(len(nums)%2 ==0):
            size = len(nums)/2
        else:
            size = len(nums)/2+1

        for i in nums:
            if i not in map1:
                map1[i] = 1
            else:
                map1[i] = map1[i]+1
        
        res = 0
        for key,value in map1.items():
            if(value >=size):
                res = key
        
        return res

        